core/include/vecmem/containers/details/aligned_multiple_placement.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// vecmem headers
+#include "vecmem/memory/unique_ptr.hpp"
+
+// Standard library headers
+#include <tuple>
+#include <type_traits>
+
+namespace vecmem {
+namespace details {
+/**
+ * @brief Allocation of aligned arrays of given types.
+ *
+ * This function allocates a chunk of memory containing a set of arrays of
+ * different types in a consecutive fashion. For example, let's say we want to
+ * allocate:
+ *
+ * 1. 3 ints (4 bytes each, aligned to 4 byte boundaries)
+ * 2. 2 longs (8 bytes each, aligned to 8 byte boundaries)
+ * 3. 7 shorts (2 bytes each, unaligned)
+ *
+ * This corresponds with the following evocation of this function:
+ *
+ *     aligned_multiple_placement<int, long, short>(r, 3, 2, 7)
+ *
+ * Under these rules, the function would allocate a chunk of memory that looks
+ * like this:
+ *
+ *      ┌────────────────────────────────────────────────────────────────┐
+ * Byte │╭ 00      ╭ 10      ╭ 20      ╭ 30      ╭ 40      ╭ 50      ╭ 60│
+ * Data │-----###IIIIIIIIIIII####LLLLLLLLLLLLLLLLSSSSSSSSSSSSSS###-------│
+ * Info │╰ A  ╰ B╰ C             ╰ D             ╰ E             ╰ F     │
+ *      └────────────────────────────────────────────────────────────────┘
```

Comment:
😍

Reply:
I knew you would like this :partying_face: